internal/rest/middlewares/auth.go

```diff
@@ -0,0 +1,37 @@
+package middlewares
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gofrs/uuid"
+	"github.com/shulganew/GophKeeper/internal/entities"
+	"github.com/shulganew/GophKeeper/internal/services"
+
+	"go.uber.org/zap"
+)
+
+func Auth(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		passVal := req.Context().Value(entities.CtxPassKey{})
+		pass, ok := passVal.(string)
+		if !ok {
+			zap.S().Errorln("Can't git pass key from context.")
+			h.ServeHTTP(res, req)
+			return
+		}
+		jwt, isSet := services.GetHeaderJWT(req.Header)
+
+		var userID uuid.UUID
+		var err error
+		if isSet {
```

Comment:
```suggestion
		if jwt, isSet := services.GetHeaderJWT(req.Header); isSet {
```
либо isSet нужно переименовать hasJwt. Из названия не понятно что сет, а переменная определяется на несколько строк выше, не сразу понятно к чему относится